aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,218 @@
+   * @return table name
+   */
+  @Override
+  protected String tableName() {
+    return fullName;
+  }
+
+  @Override
+  protected void doRefresh() {
+    Optional<Table> tableOptional = getGlueTable();
+    if (tableOptional.isPresent()) {
+      Table table = tableOptional.get();
+      GlueToIcebergConverter.validateTable(table, tableName());
+      String metadataLocation = table.parameters().get(METADATA_LOCATION_PROP);
+      refreshFromMetadataLocation(metadataLocation);
+    } else {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchTableException("Cannot find Glue table %s after refresh, " +
+            "maybe another process deleted it or revoked your access permission", tableName());
+      }
+    }
+  }
+
+  /**
+   * We rely on Glue's ConcurrentModificationException to detect commit conflicts.
+   * For users that need high frequency and high concurrency update,
+   * please consider requesting rate increase for Glue UpdateTable API,
+   * or use an external lock system such as DynamoDB lock.
+   * @param base old metadata
+   * @param metadata new metadata
+   */
+  @Override
+  protected void doCommit(TableMetadata base, TableMetadata metadata) {
+    String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
+    boolean exceptionThrown = true;
+    boolean isUpdate = false;
+    Table glueTable = null;
+    try {
+      Optional<Table> glueTableOptional = getGlueTable();
+      if (glueTableOptional.isPresent()) {
+        glueTable = glueTableOptional.get();
+        // If we try to create the table but the metadata location is already set, then we had a concurrent commit
+        if (base == null && glueTable.parameters().get(METADATA_LOCATION_PROP) != null) {
+          throw new AlreadyExistsException("Table already exists in Glue: %s", tableName());
+        }
+        isUpdate = true;
+        LOG.debug("Committing existing Glue table: {}", tableName());
+      } else {
+        LOG.debug("Committing new Glue table: {}", tableName());
+      }
+
+      String glueMetadataLocation = isUpdate ? glueTable.parameters().get(METADATA_LOCATION_PROP) : null;
+      String baseMetadataLocation = base != null ? base.metadataFileLocation() : null;
+      if (!Objects.equals(baseMetadataLocation, glueMetadataLocation)) {
+        throw new CommitFailedException(
+            "Base metadata location '%s' is not same as the current table metadata location '%s' for %s",
+            baseMetadataLocation, glueMetadataLocation, tableName());
+      }
+
+      // set parameters
+      Map<String, String> parameters = isUpdate ? Maps.newHashMap(glueTable.parameters()) : Maps.newHashMap();
+      updateParameters(parameters, newMetadataLocation);
+      persistGlueTable(isUpdate, parameters);
+      exceptionThrown = false;
+    } catch (ConcurrentModificationException e) {
+      throw new CommitFailedException(e, "Glue detected concurrent update to table %s", tableName());
```

Comment:
More error messages which should be in the form Cannot x because y. I think the concurrent update to table error should also include the information 

"""   * For users that need high frequency and high concurrency update,
    * please consider requesting rate increase for Glue UpdateTable API,
    * or use an external lock system such as DynamoDB lock."""
Which an end user will usually not see